src/v/coproc/event_listener.cc

```diff
@@ -137,15 +197,28 @@ ss::future<> event_listener::do_start() {
 
 ss::future<>
 event_listener::poll_topic(model::record_batch_reader::data_t& events) {
-    return _client
-      .fetch_partition(model::coprocessor_internal_tp, _offset, 64_KiB, 5s)
+    std::vector<kafka::fetch_request::partition> partitions;
+    partitions.push_back(kafka::fetch_request::partition{
+      .id = model::partition_id(0),
+      .current_leader_epoch = 0,
+      .fetch_offset = _offset});
+    std::vector<kafka::fetch_request::topic> topics;
+    topics.push_back(kafka::fetch_request::topic{
+      .name = model::coprocessor_internal_topic,
+      .partitions = std::move(partitions)});
+    kafka::fetch_request req{
+      .max_wait_time = 5s, .min_bytes = 0, .topics = std::move(topics)};
+    vassert(_client, "Client is null");
+    return _client->dispatch(std::move(req))
       .then([this, &events](kafka::fetch_response response) {
-          if (
-            (response.error != kafka::error_code::none)
-            || _abort_source.abort_requested()) {
+          if (response.error == kafka::error_code::unknown_topic_or_partition) {
+              return _client->stop().then([this] { _client = nullptr; });
```

Comment:
It might be that you want to update metadata, in case the leader has moved, and then exit out to the loop in `start()`.

Reply:
This should update metatdata. If you check out the run loop in start, it should observe that client is null and attempt to mitigate

Reply:
admittedly its a bit confusing to read. There is a nice refactor to these loops using coroutines in a subsequent PR i have coming up.

Reply:
Oh right, because `_offset` isn't updated.

Perhaps it would be clearer if the control flow was more explicit, some options:
 * Change the the `repeat` on [line 175](https://github.com/vectorizedio/redpanda/pull/709/files#diff-d7299fc4fae6347e196b46bd923f1fe0723c984d2bac19680c48cb953aae65aeR175) to `repeat_until` that checks offset and _client
 * Return a future to `bool` or `ss::stop_iteration` in case of error.

Reply:
> @graphcareful couldn't we just patch the kafka::client to do the right thing rather than this? might be just about as much work.

@senior7515 Yes the thought came to mind but I chose to do this after reading that Ben has that change on his todo list and will eventually get to it. So for now this change isn't too big and will fix CI. Thoughts?